### DISCUSSION TITLE
Would be nice to be able to set up delay for isLoading isFetching props

### DISCUSSION THREAD

**Discussion by pavel-blagodov** [TanStack/query#7524] [Ideas]
It's very common practice to show up spinner only after certain amount of time e.g. 1 second. It useful when we want to show spinner on slow connection and don't not show spinner on fast connection. We often rely on isLoading isFetching to show spinner so it would be nice if useQuery can be configured to keep current value for N seconds while fetching in progress and change isLoading isFetching props to true only after N seconds

**Comment by TkDodo** [comment]
since this is a UI concern and straightforward to implement in user-land, we'll likely not add this.
